_posts/REFLECT-91.md

```diff
@@ -0,0 +1,400 @@
+      responses:
+        200:
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
+        404:
+          description: User not found
+          content: {}
+```
+## Components
+
+The objects of the API are all descrived here. The objects defined within the components object will have no effect on the API unless they are explicitly referenced from properties outside the components object.
+
+* [`schemas`](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#schema-object), allows the definition of input and output data types. These types can be objects, but also primitives and arrays.
+* [`securitySchemes`](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#security-scheme-object), defines a security scheme that can be used by the operations. Supported schemes are HTTP authentication, an API key (either as a header, a cookie parameter or as a query parameter), OAuth2's common flows
+
+```yaml
+components:
+  schemas:
+    User:
+      type: object
+      properties:
+        id:
+          type: integer
+          format: int64
+        username:
+          type: string
+        firstName:
+          type: string
+        lastName:
+          type: string
+        email:
+          type: string
+        password:
+          type: string
+        phone:
+          type: string
+        userStatus:
+          type: integer
+          description: User Status
+          format: int32
+  securitySchemes:
+    reflectoring_auth:
+      type: oauth2
+      flows:
+        implicit:
+          authorizationUrl: http://reflectoring.swagger.io/oauth/dialog
+          scopes:
+            write:users: modify users
+            read:users: read users
+    api_key:
+      type: apiKey
+      name: api_key
+      in: header
+```
+
+We will know move on how to actually generate code.
+
+# Generating Code from an API Specification
+
+We will take a look at two approaches to generating the code:
+1. Going over to [Swagger Editor][swagger-editor] and
+2. The [OpenAPI Maven plugin][openapi-maven-plugin].
+
+## Generating Code from Swagger Editor
+
+Although that's an approach which I wouldn't take, I must introduce it. 
+So, Go over to [Swagger Editor][swagger-editor] and copy your `<specification>.yml` (or just use the predefined one) then select from the menu *Generate Server* and pick what kind of a server you would like to generate (I went with spring).
+
+As I said before, I wouldn't use this approach. But why?
+First, the code that I got generated was using Java 7 and Spring Boot 1.5.22.RELEASE (which both are quite old now). 
+Second, if you make a change on the specification (and changes happen all the time), you would have to copy-paste the files that were changed. 
+
+So now I will introduce you to the second approach.
+ 
+ ## Generating Code with OpenAPI Maven plugin
```

Comment:
There's too much going on in this section: generating the code, implementing `UserApi`, Testing with `curl`, ... please split it up into logical sub sections. This would make it so much easier to follow.